packages/ember-data/lib/adapters/json-api-adapter.js

```diff
@@ -0,0 +1,188 @@
+/**
+  @module ember-data
+*/
+
+import {
+  Adapter,
+  InvalidError
+} from "ember-data/system/adapter";
+
+import BuildURLMixin from "ember-data/adapters/build-url-mixin";
+
+var get = Ember.get;
+var forEach = Ember.ArrayPolyfills.forEach;
+
+/**
+  @class JSONAPIAdapter
+  @constructor
+  @namespace DS
+  @extends DS.Adapter
+*/
+export default Adapter.extend(BuildURLMixin, {
+  defaultSerializer: '-json-api',
+
+  coalesceFindRequests: false,
+
+  /**
+    @property host
+    @type {String}
+  */
+
+  /**
+    @property namespace
+    @type {String}
+  */
+
+  find: function(store, type, id, record) {
+    return this.ajax(this.buildURL(type.typeKey, id, record), 'GET');
+  },
+
+  findAll: function(store, type) {
+    return this.ajax(this.buildURL(type.typeKey), 'GET');
+  },
+
+  findMany: function(store, type, ids, records) {
+    var data = {
+      filter: {
+        id: ids.join(',')
+      }
+    };
+    return this.ajax(this.buildURL(type.typeKey, ids, records), 'GET', { data: data });
+  },
+
+  findQuery: function(store, type, query) {
+    return this.ajax(this.buildURL(type.typeKey), 'GET', { data: query });
+  },
+
+
+
+  findBelongsTo: function(store, record, link, relationship) {
+    var url, id, type;
+
+    if (link.resource) {
+      url = link.resource;
+    } else {
+      id = get(record, 'id');
+      type = record.constructor.typeKey;
+      url = this.buildURL(type, id);
+    }
+
+    return this.ajax(url, 'GET');
+  },
+
+  findHasMany: function(store, record, link, relationship) {
+    var url; //, id, type, host;
+
+    if (link.resource) {
+      url = link.resource;
+    } else {
+      // TODO
+      return;
+    }
+
+    return this.ajax(url, 'GET');
+  },
+
+
+
+  createRecord: function(store, type, record) {
+    var serializer = store.serializerFor(type.typeKey);
+
+    var snapshot = record._createSnapshot();
+    var data = serializer.serialize(snapshot);
+
+    return this.ajax(this.buildURL(type.typeKey, null, record), "POST", { data: data });
+  },
+
+  updateRecord: function(store, type, record) {
+    var serializer = store.serializerFor(type.typeKey);
+
+    var snapshot = record._createSnapshot();
+    var data = serializer.serialize(snapshot, { includeId: true });
+
+    var id = get(record, 'id');
+
+    return this.ajax(this.buildURL(type.typeKey, id, record), "PUT", { data: data });
```

Comment:
jsonapi has changed updates to always use a `PATCH` verb http://jsonapi.org/format/#crud-updating